server/src/main/java/org/opensearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -339,7 +469,10 @@ public void onFailure(Exception e) {
             } else {
                 responseActionListener = listener;
             }
-            if (sendFullVersion || previousState.nodes().nodeExists(destination) == false) {
+            if (sendRemoteState && destination.isRemoteStatePublicationEnabled()) {
```

Comment:
Lets add a TODO, to evaluate if remote publishing should be enabled only if all nodes are on latest version instead of mixed mode.

Reply:
Added TODO. will create issue

Reply:
Issue created: https://github.com/opensearch-project/OpenSearch/issues/14113